File: .tmp_py4013/batch_13_dump/3476_maximize_profit_from_task_assignment.js
// LeetCode 3476 - Maximize Profit from Task Assignment
// https://leetcode.com/problems/maximize-profit-from-task-assignment/

var maxProfit = function(workers, tasks) {
    workers = workers.slice().sort((a, b) => a - b);
    tasks = tasks.slice().sort((a, b) => a[0] - b[0]);
    let ans = 0;
    const used = new Array(tasks.length).fill(false);
    for (const w of workers) {
        let best = -1, bi = -1;
        for (let i = 0; i < tasks.length; i++) {
            if (used[i]) continue;
            if (tasks[i][0] > w) break;
            if (tasks[i][1] > best) {
                best = tasks[i][1];
                bi = i;
            }
        }
        if (bi >= 0) {
            used[bi] = true;
            ans += best;
        }
    }
    return ans;
};

File: .tmp_rs4013/rust_check17/3476_maximize_profit_from_task_assignment.rs
struct Solution;
// LeetCode 3476 - Maximize Profit from Task Assignment
// https://leetcode.com/problems/maximize-profit-from-task-assignment/

impl Solution {
    pub fn max_profit(mut workers: Vec<i32>, mut tasks: Vec<Vec<i32>>) -> i64 {
        workers.sort_unstable();
        tasks.sort_by_key(|a| a[0]);
        let mut ans = 0i64;
        let mut used = vec![false; tasks.len()];
        for w in workers {
            let mut best = -1;
            let mut bi = -1i32;
            for i in 0..tasks.len() {
                if used[i] {
                    continue;
                }
                if tasks[i][0] > w {
                    break;
                }
                if tasks[i][1] > best {
                    best = tasks[i][1];
                    bi = i as i32;
                }
            }
            if bi >= 0 {
                used[bi as usize] = true;
                ans += best as i64;
            }
        }
        ans
    }
}

fn main() {}
